website/blog/2023-03-13-webhdfs-storage-for-backend.md

```diff
@@ -0,0 +1,119 @@
+---
+title: "WebHDFS Storage Backend for Databend"
+date: 2023-03-13
+slug: 2023-03-13-webhdfs-storage-for-backend
+tags: [databend, hadoop, hdfs, webhdfs]
+description: "Get to how to configure webhdfs storage backend!"
+authors:
+  - name: PsiACE
+    url: https://github.com/psiace
+    image_url: https://github.com/psiace.png
+---
+
+[Databend](https://github.com/datafuselabs/databend) is an open-source elastic and workload-aware modern cloud data warehouse that allows you to do blazing-fast data analytics on a variety of storage services.
+
+In this blog post, we will show you how to configure WebHDFS as a storage backend for Databend.
+
+[WebHDFS](https://hadoop.apache.org/docs/current/hadoop-project-dist/hadoop-hdfs/WebHDFS.html) is a REST API that provides HTTP access to HDFS, a popular distributed file system in the big data ecosystem. By using WebHDFS, you can avoid the dependency on Java environment and specific jar packages that are required by native HDFS client.
+
+## Step 1: Prepare HDFS Environment
+
+If you already have a deployed HDFS environment, you can skip this step. The only requirement is that WebHDFS must be enabled and accessible. Note that some public cloud providers may not support WebHDFS for their managed HDFS services.
+
+If you don't have an HDFS environment, you can easily set up a local environment for testing purposes only.
+
+```bash
+git clone https://github.com/PsiACE/databend-workshop.git
+cd databend-workshop/webhdfs
+docker-compose up
+```
+
+After that, you should be able to access `http://127.0.0.1:9870` .
+
+![](../static/img/blog/webhdfs-1.png)
+
+## Step 2: Deploy Databend
+
+Follow the instructions on databend.rs to configure WebHDFS for Databend. You need to set the endpoint and root (the path in HDFS) for WebHDFS. For users who need authentication, you can also configure delegation token (not covered in this blog).
+
+```toml
+[storage]
+type = "webhdfs"
+[storage.webhdfs]
+endpoint_url = "http://127.0.0.1:9870"
+# set your root
+root = "/analyses/databend/storage"
+# if your webhdfs needs authentication, uncomment and set with your value
+# delegation = "<delegation-token>"
+```
+
+Then deploy Databend according to your preferred method.
```

Comment:
I would suggest deleting this part. Users should follow the deployment guide if they dont know how to deploy and start. A single CMD line doesn help too much here.